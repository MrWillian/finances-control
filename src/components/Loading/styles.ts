import styled from 'styled-components/native';

export const Container = styled.View`
  display: flex;
  justify-content: center;

  width: 75%;
  height: 2%;
`;

export const ProgressBar = styled.View`
  border-radius: 50px;
  background: #FFF;

  elevation: 20;
`;

export const ProgressBarLine = styled.View`
  width: 65%;
  height: 80%;
  border-radius: 50px;
  background-color: #FFD700;
`;

export const Label = styled.Text`
  color: #FFD700;
  align-self: center;
  margin-top: 10px;
`;
