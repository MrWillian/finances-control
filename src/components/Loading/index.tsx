import React from 'react';

import { Container, ProgressBar, ProgressBarLine, Label } from './styles';

const Loading: React.FC = () => {
  return (
    <Container>
      <ProgressBar>
        <ProgressBarLine />
      </ProgressBar>
      <Label>65%</Label>
    </Container>
  );
}

export default Loading;