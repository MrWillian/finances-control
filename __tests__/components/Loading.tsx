import React from 'react';
import renderer from 'react-test-renderer';

import Loading from '../../src/components/Loading';

it('renders correctly with defaults', () => {
  const button = renderer
    .create(<Loading />)
    .toJSON();
  expect(button).toMatchSnapshot();
});
