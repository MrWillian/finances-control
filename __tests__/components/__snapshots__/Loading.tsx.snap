// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": "30%",
        "justifyContent": "space-between",
        "width": "80%",
      },
    ]
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../src/assets/splash-mcduck.png",
      }
    }
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 50,
          "elevation": 15,
          "height": "5%",
          "marginTop": 20,
          "width": "80%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#FFD700",
            "borderRadius": 50,
            "height": "80%",
            "marginTop": 1,
            "width": "65%",
          },
        ]
      }
    />
  </View>
</View>
`;
